### PR TITLE
Convert `section` to be a helper

### PR DESCRIPTION
**Pull request by smfoote**
`{#hello}yellow{/hello}` and `{@section key="hello"}yellow{/section}`
are now functionally the same

Depends on #121

### CODE REVIEW

File: src/builtinHelpers.js [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
-
 import util from './util';
 
+/**
+ * Break out the logic of whether the rendering context of a section is an Array
+ * @param {Function} body The appropriate body rendering function to be rendered with `val`.
+ * @param {*} val The value to be used to render the body
+ * @return {DocumentFragment}
+ */
+let sectionResult = function(body, val, td) {
```
Comment (prashn64) [inline src/builtinHelpers.js]: Can you explicitly call the second param "ctx"?

Comment (smfoote) [inline src/builtinHelpers.js]: Sure.

**Comment by prashn64** [comment]
Resolve conflicts then mergeable.

**Comment by smfoote** [comment]
Conflicts resolved. Consider this mergeable.

**Pull request merged**